src/bsdutils/meson.build

```diff
@@ -1,4 +1,28 @@
+code = '''#include <stdio.h>
+#include <fts.h>
+
+int main(void)
+{
+  char *const fts_args[] = { ".", NULL };
+  FTS *tree = fts_open(fts_args, FTS_SEEDOT, 0);
+  if (tree)
+    return fts_close(tree);
+  return -1;
+}'''
+
+fts_flag = ''
+
+if cc.has_header_symbol('fts.h', 'fts_open', required: false)
+  if not cc.links(code, name : 'fts link check')
+    fts_flag = '-lfts'
+    if not cc.links(code, args: fts_flag, name: 'fts link check')
+      error('You may need to install musl-fts-dev.')
+    endif
+  endif
+endif
```

Comment:
I would just use:
```meson
fts_dep = dependency('', required: false)
if not cc.links(code, name : 'fts link check')
    fts_dep = dependency('musl-fts')
endif
```

Reply:
Alternatively, drop the custom code and use this:
```meson
libfts = cc.find_library('fts', required: not cc.has_function('fts_read'))
```

Using cc.find_library is correct, *iff* fts can be provided by something that is neither a libc, nor the "musl-fts" project, which is apparently the case.